daemons/execd/execd_commands.c

```diff
@@ -1374,18 +1375,26 @@ lrmd_rsc_execute_service_lib(lrmd_rsc_t * rsc, lrmd_cmd_t * cmd)
 
     params_copy = pcmk__str_table_dup(cmd->params);
 
-    action = resources_action_create(rsc->rsc_id, rsc->class, rsc->provider,
+    action = services__create_resource_action(rsc->rsc_id, rsc->class, rsc->provider,
                                      rsc->type,
                                      normalize_action_name(rsc, cmd->action),
                                      cmd->interval_ms, cmd->timeout,
                                      params_copy, cmd->service_flags);
 
     if (!action) {
         crm_err("Failed to create action, action:%s on resource %s", cmd->action, rsc->rsc_id);
+        cmd->exec_rc = PCMK_OCF_UNKNOWN_ERROR;
         cmd->lrmd_op_status = PCMK_LRM_OP_ERROR;
         goto exec_done;
     }
 
+    if (action->rc != 0) {
+        cmd->exec_rc = action->rc;
+        cmd->lrmd_op_status = PCMK_LRM_OP_ERROR;
```

Comment:
services_handle_exec_error() will set action->status appropriately so we can grab this from there too

Reply:
The new function already called services__handle_exec_error(), you don't need to call it again, you just need to set cmd->lrmd_op_status to action->status :)

action->status could be PCMK_LRM_OP_NOT_INSTALLED, etc. (This looks like the reason for the last remaining executor regression test failures.)